Userland/Libraries/LibJS/Parser.cpp

```diff
@@ -208,9 +208,10 @@ class OperatorPrecedenceTable {
 
 constexpr OperatorPrecedenceTable g_operator_precedence;
 
-Parser::ParserState::ParserState(Lexer l)
+Parser::ParserState::ParserState(Lexer l, Program::Type program_type)
     : lexer(move(l))
-    , current_token(lexer.next())
+    // We have to ensure that even the first token produced by lexer.next() does not have HTML style comments.
+    , current_token((program_type == Program::Type::Module ? lexer.disallow_html_comments() : void(), lexer.next()))
```

Comment:
Ugly use of the comma expression here, if anyone can figure out a better way I would love to change this